.github/workflows/apps/output_integration_versions.go

```diff
@@ -0,0 +1,224 @@
+					Name:       match[1],
+					MinVersion: match[2],
+				})
+			}
+		}
+	}
+
+	if err := scanner.Err(); err != nil {
+		return nil, fmt.Errorf("error reading file: %v", err)
+	}
+
+	return modules, nil
+}
+
+func fetchLatestVersion(module string) (string, error) {
+	// Fetches latest version with `go list`
+	ctx, cancel := context.WithTimeout(context.Background(), 15*time.Second)
+	defer cancel()
+
+	cmd := exec.CommandContext(ctx, "go", "list", "-m", "-versions", module)
+	var stdout, stderr bytes.Buffer
+	cmd.Stdout = &stdout
+	cmd.Stderr = &stderr
+
+	err := cmd.Run()
+	if ctx.Err() == context.DeadlineExceeded {
+		return "", fmt.Errorf("command timed out")
+	}
+	if err != nil {
+		return "", fmt.Errorf("command error: %s", stderr.String())
+	}
+
+	versions := strings.Fields(stdout.String())
+	// fmt.Println("Versions:", versions)
+	if len(versions) < 1 {
+		return "", fmt.Errorf("no versions found for module: %s", module)
+	}
+
+	return versions[len(versions)-1], nil
+}
+func isModuleInstrumented(moduleName string, instrumentedSet map[string]struct{}) bool {
+	// whether the module has automatic tracing supported (by Orchestrion)
+	_, isInstrumented := instrumentedSet[moduleName]
+	return isInstrumented
+}
+
+func fetchAllLatestVersions(modules []ModuleVersion) []ModuleVersion {
+	// Concurrently fetches the latest version of each module.
+		
+	var wg sync.WaitGroup
+	var mu sync.Mutex
+
+	updatedModules := make([]ModuleVersion, 0, len(modules))
+
+	wg.Add(len(modules))
+	for _, mod := range modules {
+		go func(mod ModuleVersion) {
+			defer wg.Done()
+			latestVersion, err := fetchLatestVersion(mod.Name)
+			if err != nil {
+				fmt.Printf("Error fetching latest version for %s: %v\n", mod.Name, err)
+				mu.Lock()
+				updatedModules = append(updatedModules, ModuleVersion{mod.Name, mod.MinVersion, "Error", mod.isInstrumented})
+				mu.Unlock()
+				return
+			}
+
+			mu.Lock()
+			updatedModules = append(updatedModules, ModuleVersion{mod.Name, 
+				mod.MinVersion, latestVersion, mod.isInstrumented})
+			mu.Unlock()
+		}(mod)
+	}
+
+	wg.Wait()
+	return updatedModules
+}
+
+func outputVersionsAsMarkdown(modules []ModuleVersion, filePath string) error {
+	file, err := os.Create(filePath)
+	if err != nil {
+		return fmt.Errorf("error creating file: %v", err)
+	}
+	defer file.Close()
+
+	fmt.Fprintln(file, "| Dependency | Minimum Version | Maximum Version | Auto-Instrumented |")
+	fmt.Fprintln(file, "|------------|-----------------|-----------------|-----------------|")
+
+	// Sort modules by name
+	sort.Slice(modules, func(i, j int) bool {
+		return modules[i].Name < modules[j].Name
+	})
+
+	for _, mod := range modules {
+		fmt.Fprintf(file, "| %s | v%s | %s | %+v\n", mod.Name, mod.MinVersion, mod.MaxVersion, mod.isInstrumented)
+	}
+	return nil
+}
+
+func main() {
+	goModPath := "integration_go.mod" // path to integration_go.mod
+	outputPath := "supported_versions.md"
+	instrumentedSet := map[string]struct{}{
```

Comment:
This is from https://github.com/DataDog/orchestrion?tab=readme-ov-file#supported-libraries